extensions/interactions/SetInput/directives/set-input-validation.service.ts

```diff
@@ -34,13 +37,124 @@ export class SetInputValidationService {
     private baseInteractionValidationServiceInstance:
       baseInteractionValidationService) {}
 
+  private isSameSet(inputA: string[], inputB: string[]): boolean {
+    let setA = new Set(inputA);
+    let setB = new Set(inputB);
+    if (setA.size !== setB.size) {
+      return false;
+    }
+    for (let element of setA.values()) {
+      if (!setB.has(element)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private isSubset(inputA: string[], inputB: string[]): boolean {
+    // Checks if set inputA is a subset of inputB.
+    let setB = new Set(inputB);
+    return inputA.every(val => setB.has(val));
+  }
+
+  private isSameRule(ruleA: Rule, ruleB: Rule): boolean {
+    return ruleA.type === ruleB.type &&
+      this.isSameSet(<string[]>ruleA.inputs.x, <string[]>ruleB.inputs.x);
+  }
+
   // TODO(#7176): Replace 'any' with the exact type. This has been kept as
   // 'any' because 'customizationArgs' is a dict with possible underscore_cased
   // keys which give tslint errors against underscore_casing in favor of
   // camelCasing.
   getCustomizationArgsWarnings(customizationArgs: any): any[] {
-    // TODO(juansaba): Implement customization args validations.
-    return [];
+    let warningsList = [];
+
+    let buttonText = customizationArgs.buttonText &&
+      customizationArgs.buttonText.value;
+    if (!angular.isString(buttonText)) {
+      warningsList.push({
+        type: AppConstants.WARNING_TYPES.ERROR,
+        message: 'Button text must be a string.'
+      });
+    } else {
+      if (buttonText.length === 0) {
+        warningsList.push({
+          message: 'Label for this button should not be empty.',
+          type: AppConstants.WARNING_TYPES.ERROR
+        });
+      }
+    }
+
+    return warningsList;
+  }
+
+  getRedundantRuleWarnings(answerGroups: AnswerGroup[]): IWarning[] {
+    let warningsList: IWarning[] = [];
+
+    let previousRules: Array<{
+      answerGroupIndex: number,
+      ruleIndex: number,
+      rule: Rule
+    }> = [];
```

Comment:
Define this array as an interface separately from here, @nishantwrp can probably help.

Reply:
Yeah, please define an `interface` for this. Also, check if an interface for this already exists.

Reply:
Done